servicetalk-http-api/src/main/java/io/servicetalk/http/api/ServiceToStreamingService.java

```diff
@@ -29,7 +30,7 @@
     private final HttpService original;
 
     ServiceToStreamingService(final HttpService original, HttpExecutionStrategy strategy) {
-        super(DEFAULT_STRATEGY.merge(strategy));
+        super(defaultStrategy() == strategy ? DEFAULT_STRATEGY : strategy);
```

Comment:
consider putting this in a package private utility method to make it easier to detect common special cases.